src/main/scala/pl/project13/scala/akka/raft/protocol/StateMetadata.scala

```diff
@@ -74,9 +79,9 @@ private[protocol] trait StateMetadata extends Serializable {
 
     def withVoteFor(term: Term, candidate: ActorRef) = copy(votes = votes + (term -> candidate))
 
-    def forLeader: LeaderMeta        = LeaderMeta(clusterSelf, currentTerm, config)
-    def forFollower: Meta            = Meta(clusterSelf, currentTerm, config, Map.empty)
-    def forNewElection: ElectionMeta = this.forFollower.forNewElection
+    def forLeader: LeaderMeta = LeaderMeta(clusterSelf, currentTerm, config)
+    def forFollower(term: Term = currentTerm): Meta = Meta(clusterSelf, term, config, Map.empty)
```

Comment:
Rather than having a default argument here, maybe it would be cleaner to chain the methods?

e.g.

```
goto(Follower) with m.forFollower.withTerm(term)
```